miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/display/MiniAppDisplay.kt

```diff
@@ -1,22 +1,35 @@
-package com.rakuten.tech.mobile.miniapp
+package com.rakuten.tech.mobile.miniapp.display
 
+import android.content.Context
 import android.view.View
+import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.LifecycleObserver
+import com.rakuten.tech.mobile.miniapp.js.MiniAppMessageBridge
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
 
 /**
  * This represents the contract by which the host app can interact with the
  * display unit of the mini app.
- * This contract complies to Android's [LifecycleObserver] contract, and when made to observe
- * the lifecycle, it automatically clears up the view state and any services registered with.
  */
-interface MiniAppDisplay : LifecycleObserver {
+interface MiniAppDisplay {
+    val basePath: String
+    val appId: String
+    val miniAppMessageBridge: MiniAppMessageBridge
 
     /**
      * Provides the view associated with the mini app to the caller for showing the mini app.
+     * [LifecycleObserver] as optional. When made to observe the lifecycle,
+     * it automatically clears up the view state and any services registered with.
      * @return [View] as mini app's view with [LayoutParams] set to match
      * the parent's dimensions.
      */
-    fun getMiniAppView(): View
```

Comment:
Is there a way we can keep the old `getMiniAppView` method as well as well and deprecate it? I guess it would keeping a context passed from the Application, so things would get complicated.

Maybe we can release a major version for the next release? Then we don't have to worry as much about minor changes to the interface.